runtime/src/transaction/tree.rs

```diff
@@ -17,13 +17,17 @@ use crate::{
 // NOTE: This should be kept in sync with go/runtime/transaction/transaction.go.
 
 #[derive(Debug)]
+#[repr(u8)]
 enum ArtifactKind {
-    Input,
-    Output,
+    _Invalid = 0,
+    Input = ARTIFACT_KIND_INPUT,
+    Output = ARTIFACT_KIND_OUTPUT,
 }
 
-const ARTIFACT_KIND_INPUT: u8 = 0;
-const ARTIFACT_KIND_OUTPUT: u8 = 1;
+// Workaround because rust doesn't support `as u8` inside match arms.
+// See https://github.com/rust-lang/rust/issues/44266
+const ARTIFACT_KIND_INPUT: u8 = 1;
```

Comment:
At least use `ArtifactKind::Input as u8` here (the workaround is only for match arms).